### PR TITLE
Feature/splashactivity

### PR DESCRIPTION
**Pull request by AnaBolovicTyllo**

### CODE REVIEW

File: app/src/main/java/com/example/movieapp/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        MovieLocalDataSource()
+    }
+    viewModel {
+        SplashViewModel(get())
+    }
+    viewModel {
+        MainViewModel(get())
+    }
+    single<String> (named("BASE_URL")) {
+        "https://api.themoviedb.org/"
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/AppModule.kt:29]: There isn't really a need for a hard coded url string as a dependency.
Try providing it in a singleton (object) `Constants.kt` instead, or even better grandle build config(enviroment) variable.

File: app/src/main/java/com/example/movieapp/AppModule.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    single<String> (named("POSTER_BASE_URL")) {
+        "https://image.tmdb.org/t/p/w600_and_h900_bestv2"
+    }
+    single {
+        OkHttpClient.Builder().apply {
+            addInterceptor(
+                Interceptor { chain ->
+                    val builder = chain.request().newBuilder()
+                    builder.header("Authorization", "Bearer 76422ff67ea5c659115400f1f574c025")
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/AppModule.kt:39]: Same here, don't expose your public api key.

Comment (luka-lukovic-tyllo) [reply app/src/main/java/com/example/movieapp/AppModule.kt:39]: Very dangerous!

File: app/src/main/java/com/example/movieapp/data/MovieRepository.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.example.movieapp.models.MovieDto
+
+class MovieRepository (
+    private val localDataSource: MovieLocalDataSource,
+    private val remoteDataSource: MovieRemoteDataSource
+) {
+
+    suspend fun loadPopularMovies(page: Int = 0): List<MovieDto> {
+        return localDataSource.getPopular(page) ?: run {
+            val response = remoteDataSource.getsPopular(page = page, apiKey = "76422ff67ea5c659115400f1f574c025")
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/data/MovieRepository.kt:13]: Extract api key to a constant or enviroment variable.

see comment above.

Comment (luka-lukovic-tyllo) [reply app/src/main/java/com/example/movieapp/data/MovieRepository.kt:13]: https://stackoverflow.com/questions/35722904/saving-the-api-key-in-gradle-properties

If you don't succeed at this just put it in a `Constants.kt` file, and add it to .gitignore

File: app/src/main/java/com/example/movieapp/models/MovieDto.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @SerializedName("vote_count")
+    val voteCount: Int? = null
+)
+
+object MovieMapper {
+    fun map(dto: MovieDto): MovieUi{
+        return MovieUi(
+            title = dto.title ?: "Unknown",
+            posterPath = dto.posterPath,
+            voteAverage = dto.voteAverage?.times(10)?.toInt()?.toDouble()?.div(10)?.toBigDecimal()?.setScale(1, RoundingMode.DOWN)?.toDouble()
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/models/MovieDto.kt:43]: Extract this into a private member function

File: app/src/main/java/com/example/movieapp/data/MovieRepository.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import android.util.Log
+import com.example.movieapp.models.MovieDto
+
+class MovieRepository (
+    private val localDataSource: MovieLocalDataSource,
+    private val remoteDataSource: MovieRemoteDataSource
+) {
+
+    suspend fun loadPopularMovies(page: Int = 0): List<MovieDto> {
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/data/MovieRepository.kt:11]: Do not overuse default paramater in functions, are they used?
Here the default is `0` but in remote and local data source its `1`.

ViewModel also keeps the `currPage` as default value of 1

File: app/src/main/java/com/example/movieapp/models/MovieDto.kt (hunk 4 above)
Comment (luka-lukovic-tyllo) [reply app/src/main/java/com/example/movieapp/models/MovieDto.kt:43]: private fun Float.roundDecimal(rounding: Int = 2): Float {
    val rounding = 10f.pow(rounding)
    return ( this * rounding ).roundToInt() / rounding
}

File: app/src/main/java/com/example/movieapp/ui/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        lifecycleScope.launch {
+            viewModel.stateFlow.collect { data ->
+                adapter.add(data)
+            }
+        }
+    }
+
+    private fun loadNextPage() {
+        viewModel.loadNextPage()
+    }
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/ui/MainActivity.kt:38]: This function is not really needed

File: app/src/main/java/com/example/movieapp/ui/MainViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    private fun loadMovies() {
+        viewModelScope.launch(Dispatchers.IO) {
+            isLoading = true
+
+            val list = movieRepository.loadPopularMovies(currentPage).map { movieDto ->
+                MovieMapper.map(movieDto)
+            }
+            _stateFlow.emit(list)
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/ui/MainViewModel.kt:38]: you should use `_state.update {} ` for updating the state of a `MutableStateFlow`

_state: MutableStateFlow = MutableStateFlow(123)
state = _stateFlow.asStateFlow()


// to update
_state.update {
    // new value
    151
}

data class Person(
    val name: String = "Luka",
    val lastName: String = "Lukovic"
)
class s {
    
    
    private val _state = MutableStateFlow(Person())
    private val state = _state.asStateFlow()
    
    init {
        _state.update { person -> 
            person.copy(
                name = "Ana",
                lastName = "Bolovic"
            )
        }
    }
}

File: app/src/main/java/com/example/movieapp/ui/MainViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    }
+
+    fun loadNextPage() {
+        viewModelScope.launch(Dispatchers.IO) {
+            currentPage++
+            val movieDtos = movieRepository.loadPopularMovies(currentPage)
+            val listOfPopular = movieDtos.map { movieDto -> MovieMapper.map(movieDto) }
+            withContext(Dispatchers.Main) {
+                _stateFlow.emit(listOfPopular)
```
Comment (luka-lukovic-tyllo) [inline app/src/main/java/com/example/movieapp/ui/MainViewModel.kt:50]: same here

File: app/src/main/java/com/example/movieapp/AppModule.kt (hunk 1 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/AppModule.kt:29]: Done

File: app/src/main/java/com/example/movieapp/AppModule.kt (hunk 2 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/AppModule.kt:39]: Done

File: app/src/main/java/com/example/movieapp/data/MovieRepository.kt (hunk 3 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/data/MovieRepository.kt:13]: Done

File: app/src/main/java/com/example/movieapp/models/MovieDto.kt (hunk 4 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/models/MovieDto.kt:43]: Done

File: app/src/main/java/com/example/movieapp/data/MovieRepository.kt (hunk 5 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/data/MovieRepository.kt:11]: Done

File: app/src/main/java/com/example/movieapp/ui/MainActivity.kt (hunk 6 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/ui/MainActivity.kt:38]: Done

File: app/src/main/java/com/example/movieapp/ui/MainViewModel.kt (hunk 7 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/ui/MainViewModel.kt:38]: Done

File: app/src/main/java/com/example/movieapp/ui/MainViewModel.kt (hunk 8 above)
Comment (AnaBolovicTyllo) [reply app/src/main/java/com/example/movieapp/ui/MainViewModel.kt:50]: Done

**Pull request merged**